File: pset6/cash.py
from cs50 import get_float
import math

def main():

    while True:
        #get number from user
        changeOwed = get_float("Change owed: ")
        #check if valid
        if changeOwed > 0:
            break

    totalCents = changeOwed * 100
    finalResult = calculateCoinCount(totalCents)
    print("The total number of coins is: ", finalResult)

def calculateCoinCount(totalCents):
    leftOverCents = totalCents
    coinCount = 0

    coinCount = math.floor(leftOverCents / 25)
    leftOverCents = leftOverCents % 25

    if leftOverCents == 0:
        return coinCount


    coinCount = math.floor(leftOverCents / 10) + coinCount
    leftOverCents = leftOverCents % 10

    if leftOverCents == 0:
        return coinCount


    coinCount = math.floor(leftOverCents / 5) + coinCount
    leftOverCents = leftOverCents % 5

    if leftOverCents == 0:
        return coinCount


    coinCount += leftOverCents


    return coinCount


if __name__ == "__main__":
    main()

File: pset1/cash.c
//Abstractions and scope
#include <cs50.h>
#include <stdio.h>
#include <math.h>

float getChangeOwed(string prompt);
int calculateCoinCount(int cents);



int main(void)
{

   float totalChange = getChangeOwed("Change owed: ");

    int cents = round(totalChange * 100);

    int coinCount = calculateCoinCount(cents);
    printf("The total number of coins is: %i\n", coinCount);


}

 // Prompt user for a positive floating point
float getChangeOwed (string prompt)
{
    float change;
    do {
        change = get_float("%s", prompt);
    }
    while (change < 0);
    return change;
}


  // Calculate number of coins
int calculateCoinCount (int p_totalCents) {
    int leftOverCents = p_totalCents;
    int coinCount;



    coinCount = floor(leftOverCents/25);
       leftOverCents = leftOverCents%25;


    if (leftOverCents == 0) {
        return coinCount;
    }

     coinCount = floor(leftOverCents/10) + coinCount;
       leftOverCents = leftOverCents%10;

    if (leftOverCents == 0) {
        return coinCount;
    }

     coinCount = floor(leftOverCents/5) + coinCount;
       leftOverCents = leftOverCents%5;

    if (leftOverCents == 0) {
        return coinCount;
    }

     coinCount = floor(leftOverCents/1) + coinCount;
       leftOverCents = leftOverCents%1;


        return coinCount;


  }
